closed/autoconf/custom-hook.m4

```diff
@@ -313,6 +314,41 @@ AC_DEFUN_ONCE([OPENJ9_PLATFORM_SETUP],
   AC_SUBST(OPENJ9_LIBS_SUBDIR)
 ])
 
+
+AC_DEFUN_ONCE([OPENJ9_CHECK_NASM_VERSION],
+[
+  OPENJ9_PLATFORM_EXTRACT_VARS_FROM_CPU($host_cpu)
+  
+  if test "x$OPENJ9_CPU" = xx86-64 ; then
+    AC_CHECK_PROG(NASM_INSTALLED,nasm,yes,no)
+    if test "x$NASM_INSTALLED" = xyes ; then
+      AC_MSG_CHECKING([whether nasm version reqirement is met])
+      # NASM version string is of the following format:
+      #  ---
+      #  NASM version 2.14.02 compiled on Dec 27 2018
+      #  ---
+      # Some builds may not contain any text after the version number
+      NASM_VERSION_STRING=`nasm -v | $SED -e 's/^.* \(@<:@0-2@:>@\.@<:@0-9@:>@@<:@0-9@:>@\.@<:@0-9@:>@@<:@0-9@:>@\).*$/\1/'`
```

Comment:
It would be clearer if this was expressed within an extra pair of brackets (with a comment):
```
[NASM_VERSION_STRING=`nasm -v | $SED -e 's/^.* \([0-2]\.[0-9][0-9]\.[0-9][0-9]\).*$/\1/'`]
```
Why limit the first component `[0-2]`? Shouldn't it be `[2-9]`?

On the other hand, `NASM_VERSION_STRING` is only used if there's a problem: maybe we could use `nasm -v` directly in the error message?